src/charm.py

```diff
@@ -263,7 +263,10 @@ def tempo_otlp_http_endpoint(self) -> Optional[str]:
         # the charm container and the tempo workload container have apparently the same
         # IP, so we can talk to tempo at localhost.
         # TODO switch to HTTPS once SSL support is added
-        return f"http://localhost:{self.tempo.receiver_ports['otlp_http']}"
+        if self.tempo.is_ready():
+            return f"http://localhost:{self.tempo.receiver_ports['otlp_http']}"
+        else:
+            return None
```

Comment:
Just a matter of taste, the `else` it's not necessary 

```suggestion
        if self.tempo.is_ready():
            return f"http://localhost:{self.tempo.receiver_ports['otlp_http']}"

        return None
```